exec/jdbc/src/main/java/org/apache/drill/jdbc/impl/DrillConnectionImpl.java

```diff
@@ -362,12 +364,33 @@ public PreparedStatement prepareStatement(String sql, int resultSetType,
                                             int resultSetHoldability) throws SQLException {
     throwIfClosed();
     try {
-      DrillPrepareResult prepareResult = new DrillPrepareResult(sql);
+      DrillRpcFuture<CreatePreparedStatementResp> respFuture = client.createPreparedStatement(sql);
+
+      CreatePreparedStatementResp resp;
+      try {
+        resp = respFuture.get();
+      } catch (InterruptedException e) {
+        // Preserve evidence that the interruption occurred so that code higher up
+        // on the call stack can learn of the interruption and respond to it if it
+        // wants to.
+        Thread.currentThread().interrupt();
+
+        throw new SQLException( "Interrupted", e );
```

Comment:
This is a create prepared statement query. Even if the client receives a message after the interrupt, the response is going to be a protobuf message (no DrillBufs). The query that is executed on client side (`LIMIT 0` query) has definitive timeout and result buffers are already released on server side (see `UserClientConnectionWrapper` in `PreparedStatementProvider`), before preparing a response message to create prepare statement request.